src/Components/Address/AddressFormFields.tsx

```diff
@@ -95,6 +117,12 @@ export const AddressFormFields = <V extends FormikContextWithAddress>(
     contextPageOwnerId: contextPageOwnerId || "",
   }
 
+  const phoneInputType: "legacy" | "rich" | null = props.withLegacyPhoneInput
+    ? "legacy"
+    : props.withPhoneNumber
+      ? "rich"
+      : null
+
```

Comment:
I've found logical assignment works well with use memo or an IIFE (besides, we should avoid double ternaries like this in particular): 

```tsx
const phoneInputType = useMemo(() => {
  if (props.withLegacyPhoneInput) {
    return 'legacy'
  }
  if (props.withPhoneNumber) {
    return 'rich'
  }
  
  return null
}, [props.withLegacyPhoneInput])
```